homeassistant/components/sonos/speaker.py

```diff
@@ -246,7 +246,7 @@ def setup(self) -> None:
     async def async_handle_new_entity(self, entity_type: str) -> None:
         """Listen to new entities to trigger first subscription."""
         self._platforms_ready.add(entity_type)
-        if self._platforms_ready == PLATFORMS and not self._subscriptions:
+        if self._platforms_ready == PLATFORMS:
```

Comment:
We should change it back.
```
2021-06-05 17:24:03 ERROR (MainThread) [homeassistant.util.logging] Exception in async_handle_new_entity when dispatching 'sonos_entity_created-RINCON_949F3E7264D201400': ('switch',)
Traceback (most recent call last):
  File "/Users/schneider/codes/personal/hass/homeassistant/components/sonos/speaker.py", line 251, in async_handle_new_entity
    await self.async_subscribe()
  File "/Users/schneider/codes/personal/hass/homeassistant/components/sonos/speaker.py", line 309, in async_subscribe
    raise RuntimeError(
RuntimeError: Attempted to attach subscriptions to player: <SoCo object at ip 192.168.178.29> when existing subscriptions exist: [<pysonos.events_asyncio.Subscription object at 0x7f98b538c370>, <pysonos.events_asyncio.Subscription object at 0x7f98b538ca00>, <pysonos.events_asyncio.Subscription object at 0x7f98b538c640>, <pysonos.events_asyncio.Subscription object at 0x7f98b538c100>, <pysonos.events_asyncio.Subscription object at 0x7f98b538c340>, <pysonos.events_asyncio.Subscription object at 0x7f98b51b7340>]
```

Reply:
When did this happen? When an alarm was created during runtime?

Reply:
yes